packages/design-system/src/components/FilterTag/FilterTag.tsx

```diff
@@ -0,0 +1,104 @@
+import { ClearIcon } from '../ClearIcon';
+import React from 'react';
+import classNames from 'classnames';
+
+const actionableKeys = ['Enter', 'Backspace', 'Delete'];
+const idFromValue = (value: string) => `${value}-tag`;
+
+export interface FilterTagProps {
+  /**
+   * Id for filter tag button. If not provided, a unique id will be automatically generated and used.
+   */
+  id?: string;
+  /**
+   * Class to be applied to the outer `<div>` that contains filter tag button.
+   */
+  className?: string;
+  /**
+   * Text for the filter tag
+   */
+  label: string;
+  /**  
+   *
+   Value for filter tag 
+   */
+  value: string;
+  /**
+   *  For screenreaders, text to read for removal
+   */
+  removeText: string;
```

Comment:
I hadn't done it in the pet-shared-components version, but this is technically only necessary when `readOnly` is true. This is a good opportunity to correct this. Maybe something like
```ts
type RemoveProps =
  | { readOnly: true; removeText: string; }
  | { readOnly?: false; removeText?: string; }; 
```
I think that'll work. I tested it [here](https://www.typescriptlang.org/play?#code/C4TwDgpgBAShC2B7AbhACgJ0WAzlAvALABQUUAPlAN5QYQCGAJgPIB2ANiAFxTAYCuEANy0EKCABUIAD2A8cfAJasA5iIC+JMpRp0mbTgH4eAM3rscw0UlRTZxqAozK1UdUJIkT-VgGNgioisUABGiCEAFGBYuDxwNugxOACU1J7EmsQkYZG6DCwc3LwCEAA01uJ2clAARAAWijVuydnhEXn6habmls2tuc1AA)